packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -228,20 +228,27 @@ const Separator = ({ separator }: Pick<BreadcrumbsProps, 'separator'>) => (
 // ----------------------------------------------------------------------------
 
 const BreadcrumbsItem = React.forwardRef((props, forwardedRef) => {
-  const { children, className, ...rest } = props;
+  const { as: asProp, ...rest } = props;
 
-  const defaultAs = !!props.href ? Anchor : !!props.onClick ? 'button' : 'span';
+  const commonProps = {
+    ...rest,
+    className: cx('iui-breadcrumbs-content', props.className),
+    ref: forwardedRef,
+  };
+
+  if (
+    asProp === 'span' ||
```

Comment:
In my original suggestion this was wrapped in `String`. Not sure why you removed that but it makes the build fail.

```suggestion
    String(asProp) === 'span' ||
```

Is it possible that your IDE is not surfacing types correctly?

Reply:
Ohh, not sure why I didn't get that error before (or maybe I missed it somehow), but I am getting it now. Wrapped `asProp` with `String()` (https://github.com/iTwin/iTwinUI/pull/2055/commits/6959296eaf75bd8db79e6b61d1648d3c7740c031).

When re-testing it, I spotted a mistake where if a `as={Link}` and `href="…"` is passed in NextJS, an `Anchor` is rendered. Fixed this error (https://github.com/iTwin/iTwinUI/pull/2055/commits/68d232f50e41c7b14dd45977c2e4e60732712d20).